pyproject.toml

```diff
@@ -36,41 +36,31 @@ disable_error_code = "attr-defined"
 # style choices
 warn_no_return = "False"
 
-# TODO: Fix typing for these modules
+# Changes mypy default to ignore all errors
 [[tool.mypy.overrides]]
 module = [
-    "pytorch_lightning.callbacks.*",
-    "pytorch_lightning.core.*",
-    "pytorch_lightning.loggers.*",
-    "pytorch_lightning.loops.*",
-    "pytorch_lightning.overrides.*",
-    "pytorch_lightning.plugins.environments.*",
-    "pytorch_lightning.plugins.training_type.*",
-    "pytorch_lightning.profiler.*",
-    "pytorch_lightning.trainer.*",
-    "pytorch_lightning.distributed.*",
-    "pytorch_lightning.tuner.*",
-    "pytorch_lightning.utilities.*",
+    "pytorch_lightning.*",
 ]
 ignore_errors = "True"
 
+# Override the default for files where we would like to enable type checking
+# TODO: Bring more files into this section
 [[tool.mypy.overrides]]
 module = [
     "pytorch_lightning.callbacks.device_stats_monitor",
     "pytorch_lightning.callbacks.early_stopping",
     "pytorch_lightning.callbacks.gpu_stats_monitor",
     "pytorch_lightning.callbacks.gradient_accumulation_scheduler",
-    "pytorch_lightning.callbacks.lr_monitor",
     "pytorch_lightning.callbacks.model_summary",
     "pytorch_lightning.callbacks.progress",
     "pytorch_lightning.callbacks.pruning",
     "pytorch_lightning.callbacks.rich_model_summary",
     "pytorch_lightning.core.optimizer",
-    "pytorch_lightning.lite.*",
```

Comment:
I will take care of it. Already gone through hell (many learnings) to get this line in here. 
#10743